ml_service/pipelines/diabetes_regression_build_train_pipeline.py

```diff
@@ -57,6 +56,14 @@ def main():
 
     # Check to see if dataset exists
     if (dataset_name not in aml_workspace.datasets):
+        # This is a hack for the bootstrap script so that we handle our global
+        # find/replace of the project name gracefully.
+        try:
+            from sklearn.datasets import load_diabetes
+        except ImportError as e:
+            print("Project has already been bootstrapped, you must provide your own data.")  # NOQA: E501
```

Comment:
I wouldn't be so strong in this message. We don't know the reason of the error for sure. We just guess. I would go with something like "Failed to load diabetes dataset, perhaps the project has already ..."

The thing is that it will still rename load_diabetes into load_we_dont_call_his_name which introduces a buggy code (that we handle with this try-except). Perhaps it would make sense to move out all this load_diabetes dataset creation to a separate module (imported in this file) and exclude that module/file from "files" in replace_project_name.

Reply:
Yes, separating dataset creation is the right way to go. A quick hotfix is to call replaceprojectname on the training script to rename the specific import.

Reply:
I thought ImportError would be enough of a limited scope to avoid any weirdness. However you made me realize that this would actually be the first time that we encounter sklearn in the flow, so this could hide a dependency error. At the very least, we need to check that sklearn exists *and* the loading function doesn't.

Re: moving it out, if we're going to do the non-hacky fix, I feel that this should be generic enough that you don't need to rely on having a predefined dataset to export and rather can use a csv.